crates/bevy_ecs/src/event.rs

```diff
@@ -374,6 +365,35 @@ impl<T: Component> Events<T> {
     }
 }
 
+impl<T> std::iter::Extend<T> for Events<T> {
+    fn extend<I>(&mut self, iter: I)
+    where
+        I: IntoIterator<Item = T>,
+    {
+        let mut event_count = self.event_count;
+        let events = iter.into_iter().map(|event| {
+            let event_id = EventId {
+                id: event_count,
+                _marker: PhantomData,
+            };
+            event_count += 1;
+            EventInstance { event_id, event }
+        });
+
+        match self.state {
+            State::A => self.events_a.extend(events),
+            State::B => self.events_b.extend(events),
+        }
+
+        trace!(
+            "Events::extend() -> [{}, {})",
+            self.event_count,
+            event_count
+        );
```

Comment:
This may not be the most useful log. Maybe just the number of events sent?

Reply:
I was mostly just trying to mirror the `Events::send` log which output the event ids.

Reply:
However, I'm open to changing it if the opinion is strong.

Reply:
not strong. maybe using rust range notation rather than an interval?

Reply:
Yeah, I think this is fine. We can revisit this once we have a clearer picture of the logs we need.